helix-term/src/ui/editor.rs

```diff
@@ -339,6 +355,22 @@ impl EditorView {
                     spans.pop();
                 }
                 HighlightEvent::Source { start, end } => {
+                    let char_buf = &mut [0; 4];
+                    let space: &str = if whitespace.render.space() == WhitespaceRenderValue::All
```

Comment:
Can you move this whole block out? It's encoding the char on each Source event but I'd rather see it happens once per render. I'd also simplify this by using to_string() on the char then using a reference